src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssetEndpoints.cs

```diff
@@ -121,15 +103,39 @@ private Dictionary<string, HashSet<string>> CreateEndpointsByAssetFile()
         return null;
     }
 
-    private List<StaticWebAssetEndpoint> CreateEndpoints(List<StaticWebAsset.StaticWebAssetResolvedRoute> routes, StaticWebAsset asset, ContentTypeProvider contentTypeMappings)
+    private readonly struct ParallelWorker(
+        List<StaticWebAssetEndpoint> collectedEndpoints,
+        List<StaticWebAssetEndpoint> currentEndpoints,
+        ITaskItem[] candidateAssets,
+        Dictionary<string, HashSet<string>> existingEndpointsByAssetFile,
+        TaskLoggingHelper log,
+        ContentTypeProvider contentTypeProvider,
+        Dictionary<string, ITaskItem> assetDetails,
+        Func<string, int> testLengthResolver,
+        Func<string, DateTime> testLastWriteResolver)
     {
-        var (length, lastModified) = ResolveDetails(asset);
-        var result = new List<StaticWebAssetEndpoint>();
-        foreach (var (label, route, values) in routes)
+        public List<StaticWebAssetEndpoint> CollectedEndpoints { get; } = collectedEndpoints;
+        public List<StaticWebAssetEndpoint> CurrentEndpoints { get; } = currentEndpoints;
+        public ITaskItem[] CandidateAssets { get; } = candidateAssets;
+        public Dictionary<string, HashSet<string>> ExistingEndpointsByAssetFile { get; } = existingEndpointsByAssetFile;
+        public TaskLoggingHelper Log { get; } = log;
+        public ContentTypeProvider ContentTypeProvider { get; } = contentTypeProvider;
+        public Dictionary<string, ITaskItem> AssetDetails { get; } = assetDetails;
+        public Func<string, int> TestLengthResolver { get; } = testLengthResolver;
+        public Func<string, DateTime> TestLastWriteResolver { get; } = testLastWriteResolver;
```

Comment:
With this may fields, wouldn't the cost of copying this struct around be pretty large?

Do you think it would be an improvement to just have two fields in the struct:
* A reference to the `DefineStaticWebAssetEndpoints` task itself (for all the state shared between workers)
* The `CurrentEndpoints` (the per-worker state)
?

Reply:
I wanted to pass the state to the worker to avoid accidentally modifying concurrent state. Even with these many fields, the struct is copied once per worker thread, which is around 8,10 times, and copying a struct even this size is super fast as it's vectorized.